File: Lib/Time.lua
local FrameBegin = require("Lib.EventCenter").FrameBegin

local cls = {}

cls.Time = 0
cls.Frame = 0
cls.Delta = 1 / 30

FrameBegin:On(cls, function(_, dt)
    local f = cls.Frame + 1
    cls.Frame = f
    cls.Time = f * dt
end)

return cls

File: CSProject/libs/LuaWrapper/Time.cs
namespace LuaWrapper;

using SFLib;

[Lua(Module = "Lib.Time")]
public class Time : LuaObject
{
    public static int Frame;
    public static float Delta;
    [Lua(Name = "Time")]
    public static float CurrentTime;
    public static float CeilToNextUpdate(float timestamp) => 0f;
}